### PR TITLE
add sparse update section in fluid dist doc

### LINKED ISSUE #8996: Add Fluid sparse update in dist doc

### PR DESCRIPTION
**Pull request by Yancey1989**
Fixed #8996

### CODE REVIEW

File: doc/design/fluid_dist/parameter_server.md [hunk 1]
```diff
@@ -59,6 +59,13 @@ After converting:
      queue. It will block until the queue has the required number of
      tensors.
 
+### Sparse Update
+
+For an embedding layer, the gradient maybe be very sparse(upper 90% is zero) for each mini-batch.
```
Comment (typhoonzero) [inline doc/design/fluid_dist/parameter_server.md]: There should be a space between sparse and `(`.

Comment (typhoonzero) [reply doc/design/fluid_dist/parameter_server.md]: upper => up to

File: doc/design/fluid_dist/parameter_server.md [hunk 2]
```diff
@@ -59,6 +59,13 @@ After converting:
...
+Fluid use [SelectedRows](../selected_rows.md) to support the sparse variable. Distributed training support `Sparse Update`,
```
Comment (typhoonzero) [inline doc/design/fluid_dist/parameter_server.md]: the sparse variable => sparse variables.

Comment (typhoonzero) [reply doc/design/fluid_dist/parameter_server.md]: support => supports

File: doc/design/fluid_dist/parameter_server.md (hunk 1 above)
Comment (typhoonzero) [reply doc/design/fluid_dist/parameter_server.md]: the gradient maybe be very sparse => the gradient may have many rows containing only 0

File: doc/design/fluid_dist/parameter_server.md (hunk 2 above)
Comment (typhoonzero) [reply doc/design/fluid_dist/parameter_server.md]: Distributed training support `Sparse Update`, which sends a `SelectedRows` variable to the parameter server to run parameter updates.

File: doc/design/fluid_dist/parameter_server.md (hunk 1 above)
Comment (Yancey1989) [reply doc/design/fluid_dist/parameter_server.md]: Done.

File: doc/design/fluid_dist/parameter_server.md (hunk 2 above)
Comment (Yancey1989) [reply doc/design/fluid_dist/parameter_server.md]: Done.

File: doc/design/fluid_dist/parameter_server.md [hunk 3]
```diff
@@ -61,10 +61,13 @@ After converting:
 
 ### Sparse Update
 
-For an embedding layer, the gradient may have many rows containing only 0 for each mini-batch.
-Fluid use [SelectedRows](../selected_rows.md) to support sparse variables. Distributed training support `Sparse Update`,
-which sends a `SelectedRows` variable to the parameter server to run parameter updates.
-It would save a lot of bandwidth and make the distributed training job have better performance.
+For embedding layers, the gradient may have many rows containing only 0 when training,
+if the gradient use a dense tensor to do parameter optimization,
+it could spend unnessesary memory, slow down the calculations and waste
```
Comment (typhoonzero) [inline doc/design/fluid_dist/parameter_server.md]: unnessesary => unnecessary

File: doc/design/fluid_dist/parameter_server.md [hunk 4]
```diff
@@ -59,6 +59,17 @@ After converting:
      queue. It will block until the queue has the required number of
      tensors.
 
+### Sparse Update
+
+For embedding layers, the gradient may have many rows containing only 0 when training,
+if the gradient use a dense tensor to do parameter optimization,
```
Comment (helinwang) [inline doc/design/fluid_dist/parameter_server.md]: use -> uses.

**Pull request merged**